latex_files/Makefile

```
@@ -1,18 +1,34 @@
 SHELL = /bin/sh
+MAIN = main
+LATEX = pdflatex
 
-all: main.pdf
+all: $(MAIN).pdf findref
 
-main.pdf: *.tex *.bib
-	pdflatex main.tex
-	bibtex main
-	touch main.tex
-	pdflatex main.tex
-	pdflatex main.tex
-	pdflatex main.tex
+$(MAIN).pdf: *.tex $(MAIN).bib
+	$(LATEX) $(LATEXFLAGS) $(MAIN).tex
+	bibtex $(LATEXFLAGS) $(MAIN)
+	touch $(MAIN).tex
+	$(LATEX) $(LATEXFLAGS) $(MAIN).tex
+	$(LATEX) $(LATEXFLAGS) $(MAIN).tex
+	$(LATEX) $(LATEXFLAGS) $(MAIN).tex
 
 short: *.tex 
-	pdflatex main.tex
+	$(LATEX) $(LATEXFLAGS) $(MAIN).tex
+
+findref:
```

```suggestion
findref: all
```

Would it be useful to call out the `all` target here, as in the above?

---

No, since that is a circular dependency; I'm saying here that `findref` runs whenever you try to build `$(MAIN).pdf`.

---

Ah, I should have read the full file more closely; I see what you mean now. Thank you for explaining!